tests/scripts/generate_tls13_compat_tests.py

```diff
@@ -416,10 +420,17 @@ def main():
     args = parser.parse_args()
 
     def get_all_test_cases():
-        for cipher, sig_alg, named_group, server, client in \
-            itertools.product(CIPHER_SUITE_IANA_VALUE.keys(), SIG_ALG_IANA_VALUE.keys(),
-                              NAMED_GROUP_IANA_VALUE.keys(), SERVER_CLASSES.keys(),
+        def cycle_zip( *iters):
+            max_len=max([len(i) for i in iters])
+            cycle_iters=zip(*[itertools.cycle(i) for i in iters])
+            for _, c in zip(range(max_len),cycle_iters):
+                yield c
+
+        for combine_fields, server, client in \
```

Comment:
I am not sure what you remove here. We still want to go through all ciphersuite, group and signature algorithm triplets. We decided not to do so for HRR because there was just too many test cases then. But when no HRR the number of test cases is reasonable I think.

Reply:
Got it